From c29e0cffb2f202489b06e63284ce20c17fd46670 Mon Sep 17 00:00:00 2001
From: Hariharan Basuthkar <quic_hbasuthk@quicinc.com>
Date: Thu, 22 Feb 2024 15:29:24 +0530
Subject: [PATCH] iw: Add power mode to print_iface_handler

Display the power mode in the output of "iw dev" command.
The power mode is obtained from the driver through
NL80211_CMD_GET_INTERFACE. Power mode is applicable only for 6 GHz,
and it is not applicable for 2 GHz and 5 GHz (it will be 0).

Signed-off-by: Hariharan Basuthkar <quic_hbasuthk@quicinc.com>
---
 interface.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/interface.c
+++ b/interface.c
@@ -391,6 +391,7 @@ static void print_channel(struct nlattr
 		channel_type = nla_get_u32(tb[NL80211_ATTR_WIPHY_CHANNEL_TYPE]);
 		printf(" %s", channel_type_name(channel_type));
 	}
+
 }
 
 static int print_iface_handler(struct nl_msg *msg, void *arg)
@@ -461,6 +462,10 @@ static int print_iface_handler(struct nl
 			printf("%s\t4addr: on\n", indent);
 	}
 
+	if (tb_msg[NL80211_ATTR_6G_REG_POWER_MODE])
+		printf("%s\tPower mode: %d\n", indent,
+		       nla_get_u8(tb_msg[NL80211_ATTR_6G_REG_POWER_MODE]));
+
 	if (tb_msg[NL80211_ATTR_MLO_LINKS]) {
 		struct nlattr *link;
 		int n;
